modelforge/dataset/parameters.py

```diff
@@ -0,0 +1,100 @@
+from enum import Enum
+from typing import Optional, List, Dict
+from pydantic import BaseModel, ConfigDict, Field, model_validator
+from typing_extensions import Self
+
+
+class CaseInsensitiveEnum(str, Enum):
+    """
+    Enum class that allows case-insensitive comparison of its members.
+    """
+
+    @classmethod
+    def _missing_(cls, value):
+        for member in cls:
+            if member.value.lower() == value.lower():
+                return member
+        return super()._missing_(value)
+
+
+# To avoid having to set config parameters for each class,
+# we will just create a parent class for all the parameters classes.
+class ParametersBase(BaseModel):
+    model_config = ConfigDict(
+        use_enum_values=True,
+        arbitrary_types_allowed=True,
+        validate_assignment=True,
+        extra="forbid",
+    )
+
+
+class DataSetName(CaseInsensitiveEnum):
+    QM9 = "QM9"
+    ANI1X = "ANI1X"
+    ANI2X = "ANI2X"
+    SPICE1 = "SPICE1"
+    SPICE2 = "SPICE2"
+    SPICE1_OPENFF = "SPICE1_OPENFF"
+    PHALKETHOH = "PhAlkEthOH"
+    TMQM = "tmQM"
+
+
+class PropertiesDefinition(ParametersBase):
+    atomic_numbers: str
+    positions: str
+    E: str
+    F: Optional[str] = None
+    dipole_moment: Optional[str] = None
+    total_charge: Optional[str] = None
+
+
+class DatasetParameters(BaseModel):
+    """
+    Class to hold the dataset parameters.
+
+    Attributes
+    ----------
+    dataset_name : DataSetName
+        The name of the dataset.
+    version_select : str
+        The version of the dataset to use.
+    num_workers : int
+        The number of workers to use for the DataLoader.
```

Comment:
Probably specify num_workers means the number of cores or threads used for loading?

Reply:
I think this is most accurately the number of threads that are launched.  I'll note that testing a range of values did not produce too much of a difference (made more of a difference for the pair list precalculations which use the data loader class but still maybe 10-20%).  With the code more stable now, might be worth doing some short benchmarks.  As we discussed today, this might make more of a difference if datasets are much larger, but we also risk increasing memory usage if they are too large.